packages/components/src/checkbox-control/style.scss

```diff
@@ -5,6 +5,20 @@
 	@include break-small() {
 		--checkbox-input-size: 20px;
 	}
+
+	.components-base-control__help {
+		margin-left: calc(var(--checkbox-input-size) + #{$grid-unit-15});
+	}
+
+	.components-base-control__field {
+		// Emsure label doesn't wrap beneath checkbox
+		display: flex;
+	}
+
+	.components-checkbox-control__label {
+		// Ensure label is aligned with checkbox along X axis
+		line-height: var(--checkbox-input-size);
+	}
```

Comment:
Let's move this part out of the `.components-checkbox-control` block to avoid unnecessary specificity.